lib/auth/iam_join_test.go

```diff
@@ -0,0 +1,345 @@
+/*
+Copyright 2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package auth
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/trace"
+	"github.com/stretchr/testify/require"
+)
+
+type mockSTSClient struct {
```

Comment:
Suggestion: have a single mock client that contains an http.Response and move the JSON logic to a `responseFromAWSIdentity` method?

Something like this:

```go
goodClient := mockClient{response: responseFromAWSIdentity(myIdentity)}
badClient := mockClient{
  response: &http.Response{
    StatusCode: http.StatusForbidden,
    Body:       io.NopCloser(strings.NewReader("access denied")),
  }
}
```

Reply:
yeah good call